### ISSUE TITLE
Noise floor warning, but it's fine

### ISSUE THREAD

**Issue by CTHobbit** [ScriptTiger/ACXMaster#1]
I used the "generate noise" option and then it gave me a warning about the noise floor being too high. 

![image](https://github.com/user-attachments/assets/3371fb8c-106c-4502-a1f4-149ee29a0b53)

Processed anyway, and the noise floor on the ACX check is -81, so not sure why it's giving that warning.

**Comment by ScriptTiger** [comment]
My tool checks the noise floor a bit differently. ACX recommends that you leave at least one second of silence, and no more than 5 seconds of silence, on the front and end of each audio file. So, my tool checks the noise floor by combining that first second and last second of your audio file to get a 2-second "window" of noise floor. This is actually much more accurate than other methods because it provides a larger sample size, as well as factors in consistency from the start to the end of your file, as opposed to something like the ACX checker, which only uses a single 0.5-second window and detects your noise floor as simply being the quietest 0.5 seconds in your audio. However, if you don't have at least 1 second of room tone at the start and end of your audio, this reading will include your actual voice as part of the noise floor.

I'm not sure if you're using the latest version of the tool, but the warning that comes with that is actually the longest warning I wrote to describe exactly what I stated above. But if you're suppressing warnings, then you won't get that lengthy explanation. ACX also does not actually openly describe its parameters for detecting noise floor, as far as window size or even region of the audio where it detects it from, if it's only detecting it from the middle somewhere, or the end, or excluding certain parts from the check. I am pretty sure ACX intentionally leaves this ambiguous so people can't mimic their check using AI. Noise floor is also not part of the Audio Lab check and is usually reviewed by a human, so it's difficult to make an algorithm which can guess how any particular human will sample it. So, just using what ACX already recommends in order to retrieve a 2-second sample just to be safe was the best solution, just to make sure it's as accurate as possible. But obviously, again, not being accurate at all if that first and last second of audio are not actually noise floor, as ACX recommends.

**Comment by CTHobbit** [comment]
Right, but it's got the right header and tail, but with the -inf floor, so that's why I added in the noise. I'm not sure why it would say the floor spikes up that high

**Comment by ScriptTiger** [comment]
-24 seems about normal if your voice is making it into the sample. Does it give you a high reading even before/without applying the noise generator? Could you DM me a link on Reddit to a sample audio with the problem so I can check it out?
